### PR TITLE
[ADD] Beta implementation of fieldservice_account

### PR DESCRIPTION
**Pull request by osimallen**
Initial implementation of field service accounting functionality

#36

### CODE REVIEW

File: fieldservice_account/models/account_invoice.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        order = self.env['fsm.order'].browse(vals.get('fsm_order_id'))
+        if order:
+            if order.location_id.analytic_account_id:
+                vals['account_analytic_id'] = order.location_id.\
+                    analytic_account_id.id
+            else:
+                raise ValidationError("No analytic account" +
+                                      " set on the order's Location")
+
+        return super().create(vals)
```
Comment (max3903) [inline fieldservice_account/models/account_invoice.py]: `return super(AccountInvoiceLine, self).create(vals)`

File: fieldservice_account/models/analytic_account.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    def create(self, vals):
+        order = self.env['fsm.order'].browse(vals.get('fsm_order_id'))
+        if order:
+            if order.location_id.analytic_account_id:
+                vals['account_id'] = order.location_id.analytic_account_id.id
+            else:
+                raise ValidationError("No analytic account set " +
+                                      "on the order's Location")
+
+        return super().create(vals)
```
Comment (max3903) [inline fieldservice_account/models/analytic_account.py]: `return super(AccountAnalyticLine, self).create(vals)`

File: fieldservice_account/models/fsm_order.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            for cost in order.contractor_cost_ids:
+                order.total_cost += cost.price_unit * cost.quantity
+
+    def action_complete(self):
+        for order in self:
+            if order.contractor_cost_ids:
+                order._create_vendor_bill()
+            if order.employee_timesheet_ids:
+                order._create_customer_invoice()
+        return super().action_complete()
```
Comment (max3903) [inline fieldservice_account/models/fsm_order.py]: `return super(FSMOrder, self).action_complete()`

File: fieldservice_account/readme/CONFIGURE.rst [hunk 4]
```diff
@@ -0,0 +1,2 @@
+No configuration is necessary for this module
+
```
Comment (max3903) [inline fieldservice_account/readme/CONFIGURE.rst:2]:
```
To configure this module, you need to:

* Go to Field Service > Master Data > Locations
* Create or select a location and set their analytic account
```

File: fieldservice_account/views/fsm_order.xml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                        </field>
+                        <field name='employee' invisible='1'/>
+                        <field name='total_cost' readonly='1'/>
+                    </group>
+                </page>
+            </page>
+        </field>
+    </record>
+</odoo>
+
```
Comment (max3903) [inline fieldservice_account/views/fsm_order.xml]: Too much empty lines at EOF.

**Review by max3903 — CHANGES_REQUESTED** [review]
@osimallen Few syntaxic/formatting changes. I haven't tested, but it looks good.

File: fieldservice_account/models/account_invoice.py (hunk 1 above)
Comment (osimallen) [reply fieldservice_account/models/account_invoice.py]: Fixed in new commit

File: fieldservice_account/models/fsm_order.py (hunk 3 above)
Comment (osimallen) [reply fieldservice_account/models/fsm_order.py]: Fixed in new commit

File: fieldservice_account/readme/CONFIGURE.rst (hunk 4 above)
Comment (osimallen) [reply fieldservice_account/readme/CONFIGURE.rst:2]: Fixed in new commit

File: fieldservice_account/views/fsm_order.xml (hunk 5 above)
Comment (osimallen) [reply fieldservice_account/views/fsm_order.xml]: Fixed in new commit

**Pull request merged**